src/vs/workbench/parts/files/browser/views/explorerViewer.ts

```diff
@@ -96,8 +140,14 @@ export class FileDataSource implements IDataSource {
 					stat.addChild(modelDirStat.children[i]);
 				}
 
```

Comment:
Assuming this method and promise is run on every file, I would think this is the only area you'd need to alter in this method.  Is there a way to say:

```
if (modelDirStat.isDirectory) {
     do existing stat.addChild logic
} else if (this.enableVirtualDirectories) {
    if resolveVirtualDirectories has not been run on stat's folder yet, run it
    if(stat.isVirtualDirectory)
        add children
}
```